packages/gatsby/cache-dir/socketIo.js

```diff
@@ -88,6 +90,27 @@ export default function socketIo() {
   }
 }
 
+function savePageDataAndStaticQueries(
+  pathname,
+  pageData,
+  staticQueriesData = {}
+) {
+  const normalizedPagePath = normalizePagePath(pathname)
+
+  if (didDataChange(normalizedPagePath, pageData, pageQueryData)) {
+    pageQueryData[normalizePagePath(pathname)] = pageData
```

Comment:
```suggestion
    pageQueryData = {
      ...pageQueryData,
      [normalizedPagePath]: pageData,
    }
```
Let's create new `pageQueryData` object if something change to trigger components that use this to update and potentially re-render